NuKeeper.Update/Process/DotNetUpdatePackageCommand.cs

```diff
@@ -22,18 +22,14 @@ public DotNetUpdatePackageCommand(IExternalProcess externalProcess)
             var projectPath = currentPackage.Path.Info.DirectoryName;
             var projectFileName = currentPackage.Path.Info.Name;
             var sourceUrl = packageSource.SourceUri.ToString();
-            var sources = allSources.CommandLine("-s");
-
-            var restoreCommand = $"restore {projectFileName} {sources}";
-            await _externalProcess.Run(projectPath, "dotnet", restoreCommand, true);
 
             if (currentPackage.Path.PackageReferenceType == PackageReferenceType.ProjectFileOldStyle)
             {
                 var removeCommand = $"remove {projectFileName} package {currentPackage.Id}";
                 await _externalProcess.Run(projectPath, "dotnet", removeCommand, true);
             }
 
-            var addCommand = $"add {projectFileName} package {currentPackage.Id} -v {newVersion} -s {sourceUrl}";
+            var addCommand = $"add {projectFileName} package {currentPackage.Id} -v {newVersion} -s {sourceUrl} --no-restore";
```

Comment:
You could create 2 implementations of the interface then create a factory to return you one with --no-restore on or not, this would be a safer change to make as the implementation would then be identical